<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImgEditor</class>
 <widget class="QWidget" name="ImgEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>862</width>
    <height>458</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ImgEditor</string>
  </property>
  <property name="styleSheet">
   <string notr="true">


    #closeBtn {
    font-size: 13px;
    background-color: transparent;
    border: none;
    border-radius: 10px; /* 添加圆角 */
}
 
   #closeBtn:hover {
    background-color: #ba3939;
    color: #ffffff;  /* 设置字体颜色为白色 */
    border-radius: 10px; /* 确保悬浮时也有圆角 */
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QWidget" name="MainBox" native="true">
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="CustomGraphicsView" name="graphicsView"/>
      </item>
      <item>
       <widget class="QDockWidget" name="dockWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>212</width>
          <height>462</height>
         </size>
        </property>
        <property name="features">
         <set>QDockWidget::DockWidgetFeature::DockWidgetFloatable|QDockWidget::DockWidgetFeature::DockWidgetMovable</set>
        </property>
        <property name="windowTitle">
         <string notr="true"/>
        </property>
        <widget class="QWidget" name="dockWidgetContents">
         <layout class="QGridLayout" name="gridLayout_2">
          <property name="leftMargin">
           <number>6</number>
          </property>
          <property name="rightMargin">
           <number>6</number>
          </property>
          <item row="0" column="0">
           <widget class="QWidget" name="Operation" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QWidget" name="Preview" native="true">
               <layout class="QVBoxLayout" name="verticalLayout_4">
                <property name="spacing">
                 <number>0</number>
                </property>
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <property name="topMargin">
                 <number>0</number>
                </property>
                <property name="rightMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QWidget" name="widget_2" native="true">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <layout class="QGridLayout" name="gridLayout">
                   <property name="leftMargin">
                    <number>0</number>
                   </property>
                   <property name="topMargin">
                    <number>4</number>
                   </property>
                   <property name="rightMargin">
                    <number>0</number>
                   </property>
                   <property name="bottomMargin">
                    <number>0</number>
                   </property>
                   <property name="spacing">
                    <number>0</number>
                   </property>
                   <item row="1" column="3">
                    <widget class="CustomToolButton" name="rBtn">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                       <horstretch>0</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                     <property name="minimumSize">
                      <size>
                       <width>30</width>
                       <height>25</height>
                      </size>
                     </property>
                     <property name="text">
                      <string/>
                     </property>
                     <property name="icon">
                      <iconset resource="assets.qrc">
                       <normaloff>:/imageEditor/右.png</normaloff>:/imageEditor/右.png</iconset>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="1">
                    <widget class="CustomToolButton" name="lBtn">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                       <horstretch>0</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                     <property name="minimumSize">
                      <size>
                       <width>30</width>
                       <height>25</height>
                      </size>
                     </property>
                     <property name="text">
                      <string/>
                     </property>
                     <property name="icon">
                      <iconset resource="assets.qrc">
                       <normaloff>:/imageEditor/左.png</normaloff>:/imageEditor/左.png</iconset>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="2">
                    <widget class="CustomToolButton" name="upBtn">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                       <horstretch>0</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                     <property name="minimumSize">
                      <size>
                       <width>0</width>
                       <height>25</height>
                      </size>
                     </property>
                     <property name="text">
                      <string/>
                     </property>
                     <property name="icon">
                      <iconset resource="assets.qrc">
                       <normaloff>:/imageEditor/上.png</normaloff>:/imageEditor/上.png</iconset>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="0">
                    <spacer name="horizontalSpacer">
                     <property name="orientation">
                      <enum>Qt::Orientation::Horizontal</enum>
                     </property>
                     <property name="sizeHint" stdset="0">
                      <size>
                       <width>0</width>
                       <height>0</height>
                      </size>
                     </property>
                    </spacer>
                   </item>
                   <item row="2" column="2">
                    <widget class="CustomToolButton" name="downBtn">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                       <horstretch>0</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                     <property name="minimumSize">
                      <size>
                       <width>0</width>
                       <height>25</height>
                      </size>
                     </property>
                     <property name="text">
                      <string/>
                     </property>
                     <property name="icon">
                      <iconset resource="assets.qrc">
                       <normaloff>:/imageEditor/下.png</normaloff>:/imageEditor/下.png</iconset>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="4">
                    <spacer name="horizontalSpacer_2">
                     <property name="orientation">
                      <enum>Qt::Orientation::Horizontal</enum>
                     </property>
                     <property name="sizeHint" stdset="0">
                      <size>
                       <width>0</width>
                       <height>0</height>
                      </size>
                     </property>
                    </spacer>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QWidget" name="widget" native="true">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <layout class="QHBoxLayout" name="horizontalLayout_4">
                   <item>
                    <widget class="InfiniteDial" name="view_scale">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                       <horstretch>0</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                     <property name="minimumSize">
                      <size>
                       <width>120</width>
                       <height>120</height>
                      </size>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QWidget" name="Trans" native="true">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_2">
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <property name="rightMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QWidget" name="widget_3" native="true">
                  <layout class="QGridLayout" name="gridLayout_3">
                   <property name="leftMargin">
                    <number>0</number>
                   </property>
                   <property name="rightMargin">
                    <number>0</number>
                   </property>
                   <item row="4" column="1">
                    <widget class="QSlider" name="rotate">
                     <property name="minimum">
                      <number>-180</number>
                     </property>
                     <property name="maximum">
                      <number>180</number>
                     </property>
                     <property name="singleStep">
                      <number>0</number>
                     </property>
                     <property name="orientation">
                      <enum>Qt::Orientation::Horizontal</enum>
                     </property>
                    </widget>
                   </item>
                   <item row="4" column="0">
                    <widget class="QLabel" name="rotateLabel">
                     <property name="text">
                      <string>旋转：</string>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QWidget" name="Step" native="true">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout">
                <property name="leftMargin">
                 <number>3</number>
                </property>
                <property name="rightMargin">
                 <number>3</number>
                </property>
                <item>
                 <widget class="QPushButton" name="undoBtn">
                  <property name="text">
                   <string/>
                  </property>
                  <property name="icon">
                   <iconset resource="assets.qrc">
                    <normaloff>:/imageEditor/undo_yes.png</normaloff>:/imageEditor/undo_yes.png</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="redoBtn">
                  <property name="text">
                   <string/>
                  </property>
                  <property name="icon">
                   <iconset resource="assets.qrc">
                    <normaloff>:/imageEditor/redo_yes.png</normaloff>:/imageEditor/redo_yes.png</iconset>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="OKBtn">
               <property name="text">
                <string>OK</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CustomGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>customgraphicsview.h</header>
  </customwidget>
  <customwidget>
   <class>InfiniteDial</class>
   <extends>QDial</extends>
   <header>infinitedial.h</header>
  </customwidget>
  <customwidget>
   <class>CustomToolButton</class>
   <extends>QToolButton</extends>
   <header>customtoolbutton.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="assets.qrc"/>
 </resources>
 <connections/>
</ui>
